tools/roxvet/analyzers/dontcompareproto/analyzer.go

```diff
@@ -0,0 +1,124 @@
+package donotcompareproto
+
+import (
+	"fmt"
+	"go/ast"
+	"go/types"
+	"strings"
+
+	"github.com/stackrox/rox/pkg/set"
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/ast/inspector"
+	"golang.org/x/tools/go/types/typeutil"
+)
+
+const doc = `Inspect calls to Equal for proto arguments that should be compared with protocompat.Equal instead`
+
+// Analyzer is the go vet entrypoint
+var Analyzer = &analysis.Analyzer{
+	Name:     "donotcompareproto",
+	Doc:      doc,
+	Requires: []*analysis.Analyzer{inspect.Analyzer},
+	Run:      run,
+}
+
+var (
+	protoPkgs = []string{
+		"*github.com/stackrox/scanner/generated",
+		"*github.com/stackrox/rox/generated/",
+	}
+
+	replacements = map[string]string{
+		"":              "Equal",
+		"[]":            "SlicesEqual",
+		"map[string][]": "MapSliceEqual",
+		"map[string]":   "MapEqual",
```

Comment:
Does this cover all cases or do we have more convoluted proto containers ?

Reply:
We have more:
```bash
# github.com/stackrox/rox/central/compliance/data
# map[string]map[string]*compliance.ComplianceStandardResult
central/compliance/data/repository_test.go:88:10: Do not use (*github.com/stretchr/testify/assert.Assertions).Equal on proto.Message
# github.com/stackrox/rox/central/graphql/resolvers
# map[*storage.ComplianceAggregation_Result]*storage.ComplianceDomain
central/graphql/resolvers/compliance_test.go:66:10: Do not use (*github.com/stretchr/testify/assert.Assertions).Equal on proto.Message
central/graphql/resolvers/compliance_test.go:76:10: Do not use (*github.com/stretchr/testify/assert.Assertions).Equal on proto.Message
central/graphql/resolvers/compliance_test.go:87:10: Do not use (*github.com/stretchr/testify/assert.Assertions).Equal on proto.Message
central/graphql/resolvers/compliance_test.go:97:10: Do not use (*github.com/stretchr/testify/assert.Assertions).Equal on proto.Message
```
But I think it's easier to refactor this test to a for from the list instead of adding another assertion